### PR TITLE
op-challenger: Only respond to claims at levels we disagree with

### PR DESCRIPTION
**Pull request by trianglesphere**
**Description**

This is important to ensure that the challenger would not counter claims that itself made. This also appears to fix the step transaction revert issue.

**Tests**

Unit tests pass. Tested with full game script.

**Metadata**

- Fixes #CLI-4194

### CODE REVIEW

File: op-challenger/fault/solver.go [hunk 1]
```diff
@@ -21,14 +21,47 @@ func NewSolver(gameDepth int, traceProvider TraceProvider) *Solver {
...
 	return s.handleMiddle(claim)
 }
 
+func (s *Solver) handleRoot(claim Claim) (*Claim, error) {
+	agree, err := s.agreeWithClaim(claim.ClaimData)
+	if err != nil {
+		return nil, err
+	}
+	// Attack the root claim if we do not agree with it
+	// Note: We always do not agree with the level
```
Comment (ajsutton) [inline op-challenger/fault/solver.go]: I'm not sure I understand what the Note part of this is trying to say.

File: op-challenger/fault/solver.go [hunk 2]
```diff
@@ -21,14 +21,47 @@ func NewSolver(gameDepth int, traceProvider TraceProvider) *Solver {
...
+	if !agree {
+		return s.attack(claim)
+	} else {
+		return nil, nil
+	}
```
Comment (ajsutton) [inline op-challenger/fault/solver.go:48]: There doesn't appear to be any tests around this code.

File: op-challenger/fault/solver.go [hunk 3]
```diff
@@ -21,14 +21,47 @@ func NewSolver(gameDepth int, traceProvider TraceProvider) *Solver {
 }
 
 // NextMove returns the next move to make given the current state of the game.
-func (s *Solver) NextMove(claim Claim) (*Claim, error) {
+func (s *Solver) NextMove(claim Claim, agreeWithClaimLevel bool) (*Claim, error) {
+	if agreeWithClaimLevel {
+		return nil, nil
+	}
```
Comment (ajsutton) [inline op-challenger/fault/solver.go:27]: I can delete this code and no tests fail. :)

**Review by ajsutton — COMMENTED** [review]
Probably worth splitting the cannon step fixes out into a separate PR. I'd want someone more familiar with cannon to review them.

**Comment by trianglesphere** [comment]
> cannon step fixes out into a separate PR

Sorry, this is a rebase issue. Will re-request when this is a clean diff

File: op-challenger/fault/solver.go (hunk 3 above)
Comment (trianglesphere) [reply op-challenger/fault/solver.go:27]: fixed.

File: op-challenger/fault/solver.go (hunk 1 above)
Comment (trianglesphere) [reply op-challenger/fault/solver.go]: Expanded this comment. It's weird corner case.

**Review by refcell — APPROVED** [review]
LGTM

**Pull request merged**